pkg/restore/restore.go

```diff
@@ -634,20 +635,36 @@ func (ctx *context) restoreResource(resource, namespace, resourcePath string) (a
 		}
 
 		// necessary because we may have remapped the namespace
-		obj.SetNamespace(namespace)
+		// if the namespace is blank, don't create the key
+		if namespace != "" {
+			obj.SetNamespace(namespace)
+		}
 
 		// add an ark-restore label to each resource for easy ID
 		addLabel(obj, api.RestoreLabelKey, ctx.restore.Name)
 
 		ctx.infof("Restoring %s: %v", obj.GroupVersionKind().Kind, obj.GetName())
-		_, err = resourceClient.Create(obj)
-		if apierrors.IsAlreadyExists(err) {
-			addToResult(&warnings, namespace, err)
+		_, restoreErr := resourceClient.Create(obj)
+		if apierrors.IsAlreadyExists(restoreErr) {
+			fromCluster, err := resourceClient.Get(obj.GetName(), metav1.GetOptions{})
+			if err != nil {
```

Comment:
I'm thinking of something like this:
```go
equal := false

if fromCluster, err := resourceClient.Get(obj.GetName(), metav1.GetOptions{}); err == nil {
  equal, err = objectsAreEqual(fromCluster, obj)
  if err != nil {
    // Log any errors trying to check equality
    ctx.infof(err.Error())
  }
} else {
  ctx.infof("error getting from-cluster...")
}

if !equal {
  e := fmt.Errorf("%s and backup does not match.", restoreErr)
  addToResult(&warnings, namespace, e)
}
```

This way, we only call `objectsAreEqual` if we successfully retrieve `fromCluster`. No matter what, if we get an error anywhere in here, it gets logged, and we add the warning in all cases unless the objects are equal. WDYT?

Reply:
Yeah, that makes sense.